Internal error when a fixture raises an exception

Using the following code in [test_import.py](https://github.com/pytest-dev/pytest-xdist/files/301452/test_import.py.txt):

``` Python
import pytest

@pytest.fixture(params=['1', '2'])
def crasher():
    raise RuntimeError

def test_aaa0(crasher):
    pass
def test_aaa1(crasher):
    pass
```

Running as `py.test -n 1 --maxfail 1` produces:
```
============================= test session starts =============================
platform win32 -- Python 3.5.1, pytest-2.9.2, py-1.4.31, pluggy-0.3.1
rootdir: E:\tmp, inifile:
plugins: xdist-1.14
gw0 [4]
scheduling tests via LoadScheduling
EINTERNALERROR> Traceback (most recent call last):
INTERNALERROR>   File "c:\users\bjones\downloads\anaconda3\lib\site-packages\_pytest\main.py", line 94, in wrap_session
INTERNALERROR>     session.exitstatus = doit(config, session) or 0
INTERNALERROR>   File "c:\users\bjones\downloads\anaconda3\lib\site-packages\_pytest\main.py", line 125, in _main
INTERNALERROR>     config.hook.pytest_runtestloop(session=session)
INTERNALERROR>   File "c:\users\bjones\downloads\anaconda3\lib\site-packages\_pytest\vendored_packages\pluggy.py", line 724, in __call__
INTERNALERROR>     return self._hookexec(self, self._nonwrappers + self._wrappers, kwargs)
INTERNALERROR>   File "c:\users\bjones\downloads\anaconda3\lib\site-packages\_pytest\vendored_packages\pluggy.py", line 338, in _hookexec
INTERNALERROR>     return self._inner_hookexec(hook, methods, kwargs)
INTERNALERROR>   File "c:\users\bjones\downloads\anaconda3\lib\site-packages\_pytest\vendored_packages\pluggy.py", line 333, in <lambda>
INTERNALERROR>     _MultiCall(methods, kwargs, hook.spec_opts).execute()
INTERNALERROR>   File "c:\users\bjones\downloads\anaconda3\lib\site-packages\_pytest\vendored_packages\pluggy.py", line 596, in execute
INTERNALERROR>     res = hook_impl.function(*args)
INTERNALERROR>   File "<remote exec>", line 49, in pytest_runtestloop
INTERNALERROR>   File "c:\users\bjones\downloads\anaconda3\lib\site-packages\execnet\gateway_base.py", line 737, in receive
INTERNALERROR>     raise self._getremoteerror() or EOFError()
INTERNALERROR> EOFError

=================================== ERRORS ====================================
_______________________ ERROR at setup of test_aaa0[1] ________________________
[gw0] win32 -- Python 3.5.1 c:\users\bjones\downloads\anaconda3\python.exe
@pytest.fixture(params=['1', '2'])
    def crasher():
>       raise RuntimeError
E       RuntimeError

test_import.py:5: RuntimeError
!!!!!!!!!!!! xdist.dsession.Interrupted: stopping after 1 failures !!!!!!!!!!!!
=========================== 1 error in 0.59 seconds ===========================```
```
Removing either the ``--maxfail`` or the `-n` flag avoids the INTERNALERROR.

--- a/CHANGELOG
+++ b/CHANGELOG
@@ -7,6 +7,9 @@
 - display progress during collection only when in a terminal, similar to pytest #1397 issue.
   Thanks Bruno Oliveira for the PR.
 
+- fix internal error message when ``--maxfail`` is used (#62, #65).
+  Thanks Collin RM Stocks and Bryan A. Jones for reports and Bruno Oliveira for the PR.
+
 
 1.14
 ----

--- a/testing/acceptance_test.py
+++ b/testing/acceptance_test.py
@@ -664,3 +664,24 @@ def test_color_yes_collection_on_non_atty(testdir, request):
     assert '\x1b[1m' in result.stdout.str()
     assert 'gw0 [10] / gw1 [10]' in result.stdout.str()
     assert 'gw0 C / gw1 C' not in result.stdout.str()
+
+
+def test_internal_error_with_maxfail(testdir):
+    """
+    Internal error when using --maxfail option (#62, #65).
+    """
+    testdir.makepyfile("""
+        import pytest
+
+        @pytest.fixture(params=['1', '2'])
+        def crasher():
+            raise RuntimeError
+
+        def test_aaa0(crasher):
+            pass
+        def test_aaa1(crasher):
+            pass
+    """)
+    result = testdir.runpytest_subprocess('--maxfail=1', '-n1')
+    result.stdout.fnmatch_lines(['* 1 error in *'])
+    assert 'INTERNALERROR' not in result.stderr.str()

--- a/xdist/remote.py
+++ b/xdist/remote.py
@@ -46,7 +46,10 @@ class SlaveInteractor:
         self.log("entering main loop")
         torun = []
         while 1:
-            name, kwargs = self.channel.receive()
+            try:
+                name, kwargs = self.channel.receive()
+            except EOFError:
+                return True
             self.log("received command", name, kwargs)
             if name == "runtests":
                 torun.extend(kwargs['indices'])

